src/Microsoft.SqlTools.ServiceLayer/DisasterRecovery/DisasterRecoveryService.cs

```diff
@@ -332,106 +342,122 @@ internal void PerformBackup(BackupOperation backupOperation)
         {
             backupOperation.PerformBackup();
         }
-        
+
         /// <summary>
-        /// Create a backup task for execution and cancellation
+        /// Create a backup script task
         /// </summary>
         /// <param name="sqlTask"></param>
         /// <returns></returns>
-        internal async Task<TaskResult> BackupTaskAsync(SqlTask sqlTask)
+        internal async Task<TaskResult> BackupScriptTaskAsync(SqlTask sqlTask)
```

Comment:
As discussed, there is no need / sense to make this 2 different methods for scripting and execution. Combine into one, have a config that lets you script, then execute, then send the result.

A good pattern from SSDT:
- Create an `EnumerateSteps` method that yield returns tasks to be executed in sequence. This would build up a set of steps (e.g. `verifyInputs`, `createPreview`, `createScript`, `executeTask`) that should be run
- Have you main method just iterate over these. For each:
  - Check cancellation status and send cancelled notification / complete the task if cancelled
  - Execute the current step in a try/catch loop
  - If an exception occurred, send notification that the task failed
  - Otherwise, continue the loop until done
Once all steps are complete, send the `TaskComplete` notification and clean up

If you do it this way, you can have identical task service implementation for Backup, Restore and most other tasks. All get the same basic config (e.g. to do scripting, execution etc.) and you just need to specialize the actions. Backup can be made much smaller and we'll reduce the code overall so that new tasks are really cheap to add.